packages/jaeger-ui/src/components/TracePage/ScrollManager.tsx

```diff
@@ -132,7 +132,7 @@ export default class ScrollManager {
     const { duration, spans, startTime: traceStartTime } = this._trace;
     const isUp = direction < 0;
     const boundaryRow = isUp ? xrs.getTopRowIndexVisible() : xrs.getBottomRowIndexVisible();
-    const spanIndex = xrs.mapRowIndexToSpanIndex(boundaryRow);
+    const spanIndex = xrs.mapRowIndexToSpanIndex(startRow != null ? startRow : boundaryRow);
```

Comment:
Is calculating `boundaryRow` even when it is not used intentional?

Reply:
I see we have `boundary` which is the index of a row and `boundaryRow` which is also the index of a row...
changed it to a `let boundaryRow` with an `if ... else if ... else` so that `xrs.get${Top/Bottom}RowIndexVisible` isn't called unnecessarily.